lib/Conversion/XTenNNToTorch.cpp

```diff
@@ -37,20 +37,25 @@ using namespace mlir::torch;
 
 namespace {
 
-Value toTorchTensorTypeCast(PatternRewriter &rewriter, Value input) {
+Type toTorchTensorTypeCast(PatternRewriter &rewriter, ShapedType ty) {
+  auto elementType = ty.getElementType();
 
-  auto tensorTy = dyn_cast<ShapedType>(input.getType());
-  auto sizes = tensorTy.getShape();
-  auto tensorEltTy = tensorTy.getElementType();
-  if (tensorEltTy.isSignlessInteger()) {
-    tensorEltTy = rewriter.getIntegerType(tensorTy.getElementTypeBitWidth(), true);
+  auto intElementType = dyn_cast<IntegerType>(ty.getElementType());
+  if (intElementType && intElementType.isSignlessInteger() && intElementType.getWidth() != 1) {
+      elementType = rewriter.getIntegerType(elementType.getIntOrFloatBitWidth(),
+        /*isSigned=*/true);
```

Comment:
Should we make `isSigned` an argument of `toTorchTensorTypeCast` to force this decision per op that is being lowered? Some ops might need unsigned

Reply:
Discussed offline.